src/java.desktop/unix/classes/sun/awt/X11GraphicsDevice.java

```diff
@@ -150,55 +150,57 @@ public String getIDstring() {
     @Override
     public GraphicsConfiguration[] getConfigurations() {
         if (configs == null) {
-            synchronized (configLock) {
-                makeConfigurations();
-            }
+            makeConfigurations();
         }
         return configs.clone();
     }
 
     private void makeConfigurations() {
-        if (configs == null) {
```

Comment:
To make the diff less distruptive you can move the ssyncronisation to the getConfigurations and getDefaultConfiguration where we have kind of DCL.

Reply:
Sure thing. Done.